### PR TITLE
docs: improve route_to() and url_to()

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
- update PHPDoc
- update User Guide
- fix sample code
- add test

**Checklist:**
- [x] Securely signed commits
- [] Component(s) with PHPDoc blocks, only if necessary or adds value
- [] Unit testing, with >80% coverage
- [x] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

File: system/Helpers/url_helper.php [hunk 1]
```diff
@@ -544,14 +544,15 @@ function mb_url_title(string $str, string $separator = '-', bool $lowercase = fa
 
 if (! function_exists('url_to')) {
     /**
-     * Get the full, absolute URL to a controller method
+     * Get the full, absolute URL to a route named or controller method
```
Comment (datamweb) [inline system/Helpers/url_helper.php:547]:
```suggestion
     * Get the full, absolute URL to a route name or controller method
```

**Comment by kenjis** [comment]
By the way, do we need `route_to()`?
I don't see there is a particular use case.

**Comment by lonnieezell** [comment]
route_to works with named routes and controller/methods instead of urls so yes I think it still needs to stay.

**Comment by kenjis** [comment]
Rebased and add a commit.

**Comment by kenjis** [comment]
Rebased and pushed 4 commits.

**Comment by kenjis** [comment]
Please review.

**Comment by kenjis** [comment]
Can anyone review this?

**Review by samsonasik — APPROVED** [review]

**Comment by kenjis** [comment]
Thanks! @samsonasik

**Pull request merged**